docs/reference/search/rrf.asciidoc

```diff
@@ -1,6 +1,423 @@
 [[rrf]]
 === Reciprocal rank fusion
 
-Reciprocal Rank Fusion (RRF) is a simple method to combine document result sets
-from multiple queries where the queries' document scores may be unrelated.
+preview::[]
 
+https://plg.uwaterloo.ca/~gvcormac/cormacksigir09-rrf.pdf[Reciprocal rank fusion (RRF)]
+is a method for combining multiple result sets with different relevance
+indicators into a single result set. RRF requires no tuning, and the different
+relevance indicators do not have to related to each other to achieve high-quality
+results.
+
+RRF uses the following formula to determine the score for ranking each document:
+
+[source,python]
+----
+score = 0.0
+for q in queries:
+    if d in result(q):
+        score += 1.0 / ( k + rank( result(q), d ) )
+return score
+
+# where
+# k is a ranking constant
+# q is a query in the set of queries
+# d is a document in the result set of q
+# result(q) is the result set of q
+# rank( result(q), d ) is d's rank within the result(q) starting from 1
+----
+// NOTCONSOLE
+
+[[rrf-api]]
+==== Reciprocal rank fusion API
+
+You can use RRF as part of a <<search-search, search>> to combine and rank
+documents using multiple result sets from
+
+* 1 query and 1 or more kNN searches
+* 2 or more kNN searches
+
+The `rrf` parameter is an optional object defined as part of a search request's
+<<request-body-rank, rank parameter>>. The `rrf` object contains the following
+parameters:
+
+`rank_constant`::
+(Optional, integer) This value determines how much influence documents in individual
+result sets per query have over the final ranked result set. A higher value indicates
+that lower ranked documents have more influence. This value must be greater than or
+equal to `1`. Defaults to `60`.
+
+`window_size`::
+(Optional, integer) This value determines the size of the individual result sets per
+query. A higher value will improve result relevance at the cost of performance. The final
+ranked result set is pruned down to the search request's <<search-size-param, size>.
+`window_size` must be greater than or equal to `size` and greater than or equal to `1`.
+Defaults to `100`.
+
+An example request using RRF:
+
+[source,console]
+----
+GET example-index/_search
+{
+    "query": {
+        "term": {
+            "text": "shoes"
+        }
+    },
+    "knn": {
+        "field": "vector",
+        "query_vector": [1.25, 2, 3.5],
+        "k": 50,
+        "num_candidates": 100
+    },
+    "rank": {
+        "rrf": {
+            "window_size": 50,
+            "rank_constant": 20
+        }
+    }
+}
+----
+// TEST[skip:example fragment]
+
+In the above example, we first execute the knn search to get its global top 50 results.
+Then we execute the query to get its global top 50 results. Afterwards, on a coordinating
+node, we combine the the knn search results with the query results and rank them based on the
+rrf method to get the final top 10 results.
```

Comment:
Should we refer to rrf with capital letters to indicate it's an acronym?

Reply:
oh well we don't do that for knn, so maybe it does not matter.

Reply:
You're right. I changed these. I prefer the consistency.